### ISSUE TITLE
Svelte 5: Rune is not stateless in SSR

### ISSUE THREAD

**Issue by J4gQBqqR** [sveltejs/kit#12507]
### Describe the bug

I use state runes in *.svelte.ts files. During SSR and hydration, my expectation is that the server side application should be stateless. To put it in another word, the server side process should not remember my last application's variable state in RAM and continue to reuse them during the next page reload.

However, my observation is different with svelte 5 rune. I found out that if I use rune in SSR, the rune state will be remembered by my node process and reuse when the webpage is refreshed (pressing F5 in the browser). This creates a lot of issues in my application as the server side code is suddenly dependent on previous state.

To illustrate the problem, I have a demo on stackblitz. In this demo, I have a class which stored three colors and pop out a color when the getter function is called:
```ts
export class ColorManager {
  private readonly initialLightColors: string[] = [
    '#8dd3c7',
    '#ffffb3',
    '#bebada',
  ];
  private availableColors: string[];

  constructor() {
    this.availableColors = [...this.initialLightColors];
  }

  get color(): string {
    if (this.availableColors.length === 0) {
      throw new Error('running out of color');
    }
    console.log(this.availableColors);

    return this.availableColors.shift()!;
  }
}

export const colorManager = $state<ColorManager>(new ColorManager());
```

When I use this colorManager rune instance in my webpage, after refreshing 3 times in my browser, my server side terminal errors out on 'running out of color'. My expectation is that each time when my browser refreshed, it will start from scratch, i.e. a new instance of colorManager will be created with 3 available colors. However, the result failed my expectation.

### Reproduction

1. cd my-app/
2. npm install
3. npx vite
4. refresh the webpage 3 times and observe the color array on the server side runs out in the console output
![image](https://github.com/user-attachments/assets/3c17f2ea-768a-4dea-b5eb-a2827d3c052b)

https://stackblitz.com/edit/vitejs-vite-4uijlv

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @sveltejs/adapter-auto: ^3.2.2 => 3.2.2 
    @sveltejs/kit: ^2.5.18 => 2.5.18 
    @sveltejs/vite-plugin-svelte: ^3 => 3.1.1 
    svelte: ^5.0.0-next.1 => 5.0.0-next.184 
    vite: ^5.3.3 => 5.3.3
```

### Severity

serious, but I can work around it

### Additional Information

I can only partially work around this issue by not throwing errors in my colorManager but to loop through the list of colors. However, this is not ideal as I will definitely face other issues when using runes in sveltekit.

Is this a side effect of runes' mechanism using object Proxy? Can sveltekit change its behavior of using runes to be stateless on its side? What can be done to mitigate this issue? I am pretty sure I will start using runes everywhere in my code after the feature is out, I do not want to deal with this side effect all along rune usage.

**Comment by Conduitry** [comment]
This isn't specific to Svelte 5. Any global variables will behave this way. https://kit.svelte.dev/docs/state-management

JS as a language does not yet provide a way to track async context so that we could automatically separate different versions of variables for different requests. Until it does and we decide how we want to expose that to users, your best bet is probably to put these sorts of things on `event.locals`.

**Comment by J4gQBqqR** [comment]
Hi, thanks for your comment. I investigated the context API method and the `event.local`. 

I have made some judgements, but not sure if my understanding is correct or not. If you do not mind, can I ask for some extra help from you to confirm if my understanding is correct or not? Thank you in advance for your help.

- If I use the context API:
If I declare a store or rune, which I want to share between components, in the highest `+layout.svelte` `<script>` section and access it in child page and child layout. There are two disadvantages:
1. I can only access it in files under /src/routes folder. I cannot access it in files under /src/lib folder. Is this understanding correct?
2. The state updated in CSR will 'flashing' during state updates during hydration, potentially causing performance issue?

To get around point 1, I can try to put all files into `/src/routes folder`. Is this a good solution? Is this what I am supposed to do when using sveltekit, i.e. try to get rid of /src/lib folder?
To get around point 2, I can try to pass state down into components rather than up as much as possible. For those situations where I have to 'pass up', I will accept the draw back of 'flashing' states during hydration. Is this correct?

- If I use the `event.locals`:
1. `event.locals` is a way to request a stateless piece of data for each request, not a way to share `$state` or `$derived` or `writable()` between components for reactivity. Is this correct?
2. For example, I cannot share a `$state` rune or a store value from `event.locals` and expect a component changing this value will trigger another component to update.
